datafusion/common/src/config.rs

```diff
@@ -408,7 +408,7 @@ config_namespace! {
         /// parquet files by serializing them in parallel. Each column
         /// in each row group in each output file are serialized in parallel
         /// leveraging a maximum possible core count of n_files*n_row_groups*n_columns.
-        pub allow_single_file_parallelism: bool, default = false
+        pub allow_single_file_parallelism: bool, default = true
```

Comment:
We could choose to leave this feature disabled by default if we wish to merge this before the next release. That would give us more time to ensure there are no other issues and improve test coverage further.

Reply:
We are about to release 35.0.0 which would not include this change. See voting thread here: https://lists.apache.org/thread/onbs8l0w5s7693fchpyvwwgh61gf1jf8

Thus I think it would actually be best to turn this default to true so the code gets maximum "bake time" from people working off `main` before a release